pkg/telemetry/context.go

```diff
@@ -5,12 +5,17 @@ import (
 )
 
 type rpcContextName string
+type dispatchThrottlingThreshold uint32
 
 const (
 	rpcInfoContextName rpcContextName = "rpcInfo"
 	Throttled          string         = "Throttled"
 )
 
+const (
+	DispatchThrottlingThreshold dispatchThrottlingThreshold = iota
```

Comment:
Please use a context getter and setter pattern. It's more idiomatic and safer Go when it comes to ctx scoped values. For more info see: https://go.dev/blog/context

```go

type dispatchThrottlingThresholdCtxKey string

func ContextWithDispatchThrottlingThreshold(ctx context.Context, threshold uint32) context.Context {
  return context.WithValue(ctx, dispatchThrottlingThresholdCtxKey, threshold)
}

func DispatchThrottlingThresholdFromContext(ctx) (uint32, bool) {
    return ctx.Value(dispatchThrottlingThresholdCtxKey).(uint32)
}
```